### PR TITLE
updated screen control setup code in ProjectGemCatalogScreen to setup…

### PR DESCRIPTION
**Pull request by tkothadev**
… ScreensCtrl based on parent of parent

Signed-off-by: T.J. Kotha <112996779+tkothadev@users.noreply.github.com>

## What does this PR do?

This PR updates `ProjectGemCatalogScreen` and `GemCatalogScreen` to call a single function `GemCatalogScreen::setupScreensControl(QWidget* parent)` to handle initialization of `m_screensControl`. The reason for this is because `ProjectGemCatalogScreen` fails to initialize that variable, because it is not a direct descendant of the `ScreenCtrls` class, but rather one of `CreateProjectCtrl` or `UpdateProjectCtrl`.

As such, I've moved all initialization of `m_screenControls` to a helper function that can be called independently to pass in the necessary parent chain.

This PR is the fix for the following issue: https://github.com/o3de/o3de/issues/13770

## How was this PR tested?

_Please describe any testing performed._

All testing was performed manually to verify that the behavior observed in the original issue was fixed.

From the Project Screen:
ProjejctGemCatalogEditButtonFixed

From the Gems Tab:
GemTabGemCatalogEditButtonFixed

When Creating a new Project:
CreateProjectGemCatalogEditButtonFixed

### CODE REVIEW

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp [hunk 1]
```diff
@@ -120,20 +117,16 @@ namespace O3DE::ProjectManager
         constexpr int minHeaderSectionWidth = 100;
         AdjustableHeaderWidget* listHeaderWidget = new AdjustableHeaderWidget(
             QStringList{ tr("Gem Image"), tr("Gem Name"), tr("Gem Summary"), tr("Status") },
-            QVector<int>{
-                GemPreviewImageWidth + AdjustableHeaderWidget::s_headerTextIndent,
-                -GemPreviewImageWidth - AdjustableHeaderWidget::s_headerTextIndent + GemItemDelegate::s_defaultSummaryStartX - 30,
-                0, // Section is set to stretch to fit
-                GemItemDelegate::s_statusIconSize + GemItemDelegate::s_statusButtonSpacing + GemItemDelegate::s_buttonWidth + GemItemDelegate::s_contentMargins.right()
-            },
+            QVector<int>{ GemPreviewImageWidth + AdjustableHeaderWidget::s_headerTextIndent,
```
Comment (tkothadev) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:120]: These indentation changes were made by the Visual Studio IDE.

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp [hunk 2]
```diff
@@ -102,7 +93,13 @@ namespace O3DE::ProjectManager
 
         m_gemInspector = new GemInspector(m_gemModel, m_rightPanelStack);
 
-        connect(m_gemInspector, &GemInspector::TagClicked, [=](const Tag& tag) { SelectGem(tag.id); });
+        connect(
```
Comment (tkothadev) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:96]: This indentation was updated by Visual Studio IDE

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp [hunk 3]
```diff
@@ -67,31 +67,22 @@ namespace O3DE::ProjectManager
 
         connect(m_gemModel, &GemModel::gemStatusChanged, this, &GemCatalogScreen::OnGemStatusChanged);
         connect(m_gemModel, &GemModel::dependencyGemStatusChanged, this, &GemCatalogScreen::OnDependencyGemStatusChanged);
-        connect(m_gemModel->GetSelectionModel(), &QItemSelectionModel::selectionChanged, this, [this]{ ShowInspector(); });
+        connect(
```
Comment (tkothadev) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:70]: this indentation was updated by visual studio ide

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp [hunk 4]
```diff
@@ -67,31 +67,22 @@ namespace O3DE::ProjectManager
...
+            m_gemModel->GetSelectionModel(),
+            &QItemSelectionModel::selectionChanged,
+            this,
+            [this]
+            {
+                ShowInspector();
+            });
```
Comment (cgalvan) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:77]: `ShowInspector` is a slot, so you could just do:
```suggestion
            connect(m_gemModel->GetSelectionModel(), &QItemSelectionModel::selectionChanged, this, &GemCatalogScreen::ShowInspector);
```

**Review by cgalvan — APPROVED** [review]

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp [hunk 5]
```diff
@@ -67,31 +67,16 @@ namespace O3DE::ProjectManager
...
-            }
-
-            ScreenWidget* editGemScreen = m_screensControl->FindScreen(ProjectManagerScreen::EditGem);
-            if (editGemScreen)
-            {
-                EditGem* editGem = static_cast<EditGem*>(editGemScreen);
-                connect(editGem, &EditGem::GemEdited, this, &GemCatalogScreen::HandleGemEdited);
-            }
-        }
+        setupScreensControl(parent);
```
Comment (forkercat) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:79]: nit: Should we use `SetUp` over `Setup`? And I think the first letter should be capitalized.

**Review by forkercat — APPROVED** [review]

File: Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp (hunk 3 above)
Comment (tkothadev) [reply Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp:70]: No that is not. I will revert it back to 1 `connect` call

**Review by AMZN-Phil — APPROVED** [review]

**Pull request merged**